### ISSUE TITLE
ApplicationSet Controller Does Not Receive Proxy Configuration

### ISSUE THREAD

**Issue by caseyscarborough** [redhat-developer/gitops-operator#249]
## Describe the bug

Our network configuration requires us to use a proxy for external URLs. When using v1.3.1 of the Red Hat Gitops Operator on Red Hat Openshift 4.7, the `openshift-gitops-applicationset-controller` does not have the global proxy configuration applied. 

This causes us to be unable to use `ApplicationSet` resources due to the following error in the `openshift-gitops-applicationset-controller` logs:

```
Error during fetching repo: `git fetch origin HEAD --tags --force` failed timeout after 1m30s
```

Here's the full log:

```
time="2021-11-30T23:19:45Z" level=error msg="error generating params" error="Error during fetching repo: `git fetch origin HEAD --tags --force` failed timeout after 1m30s" generator="&{0xc000803cd0}"
time="2021-11-30T23:19:45Z" level=error msg="error generating application from params" error="Error during fetching repo: `git fetch origin HEAD --tags --force` failed timeout after 1m30s" generator="{<nil> <nil> 0xc0005acd00 <nil> <nil> <nil>}"
2021-11-30T23:19:45.570Z	ERROR	controller-runtime.manager.controller.applicationset	Reconciler error	{"reconciler group": "argoproj.io", "reconciler kind": "ApplicationSet", "name": "my-applicationset", "namespace": "openshift-gitops", "error": "Error during fetching repo: `git fetch origin HEAD --tags --force` failed timeout after 1m30s", "errorVerbose": "`git fetch origin HEAD --tags --force` failed timeout after 1m30s
Error during fetching repo
github.com/argoproj-labs/applicationset/pkg/services.checkoutRepo
    /remote-source/app/pkg/services/repo_service.go:156
github.com/argoproj-labs/applicationset/pkg/services.(*argoCDService).GetDirectories
    /remote-source/app/pkg/services/repo_service.go:81
github.com/argoproj-labs/applicationset/pkg/generators.(*GitGenerator).generateParamsForGitDirectories
    /remote-source/app/pkg/generators/git.go:74
github.com/argoproj-labs/applicationset/pkg/generators.(*GitGenerator).GenerateParams
    /remote-source/app/pkg/generators/git.go:58
github.com/argoproj-labs/applicationset/pkg/generators.Transform
    /remote-source/app/pkg/generators/generator_spec_processor.go:52
github.com/argoproj-labs/applicationset/pkg/controllers.(*ApplicationSetReconciler).generateApplications
    /remote-source/app/pkg/controllers/applicationset_controller.go:220
github.com/argoproj-labs/applicationset/pkg/controllers.(*ApplicationSetReconciler).Reconcile
    /remote-source/app/pkg/controllers/applicationset_controller.go:93
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler
    /remote-source/deps/gomod/pkg/mod/sigs.k8s.io/controller-runtime@v0.9.0/pkg/internal/controller/controller.go:298
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem
    /remote-source/deps/gomod/pkg/mod/sigs.k8s.io/controller-runtime@v0.9.0/pkg/internal/controller/controller.go:253
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2
    /remote-source/deps/gomod/pkg/mod/sigs.k8s.io/controller-runtime@v0.9.0/pkg/internal/controller/controller.go:214
runtime.goexit
    /opt/rh/go-toolset-1.16/root/usr/lib/go-toolset-1.16-golang/src/runtime/asm_amd64.s:1371"}
```

## Expected behavior

Similar to how `openshift-gitops-server` and `openshift-gitops-dex-server` receives the `HTTP_PROXY`, `HTTPS_PROXY`, and `NO_PROXY` environment variables with no additional configuration, the `openshift-gitops-applicationset-controller` should do the same.

## Screenshots

I can see the environment configuration for the ApplicationSet Controller which is missing the proxy information, but it's not possible to change it as it reverts the changes since it is owned by the ArgoCD instance.

<img width="843" alt="Screen Shot 2021-11-30 at 6 24 45 PM" src="https://user-images.githubusercontent.com/3237612/144144174-adfffee0-2d0e-4329-98c6-4f022e67c181.png">

Comparing the applicationset-controller Deployment to the other Deployments, all Deployments have the proxy configuration except for the applicationset-controller.

<img width="1351" alt="Screen Shot 2021-11-30 at 6 26 26 PM" src="https://user-images.githubusercontent.com/3237612/144144265-bc7a0899-160b-43a7-8a13-572236b98f07.png">

<img width="1362" alt="Screen Shot 2021-11-30 at 6 45 38 PM" src="https://user-images.githubusercontent.com/3237612/144146030-2fdf8c70-41b9-48ed-abf0-6ae37b8818bc.png">

<img width="1363" alt="Screen Shot 2021-11-30 at 6 46 29 PM" src="https://user-images.githubusercontent.com/3237612/144146092-dee4ebef-6f39-41c3-a6b2-e94fa26d59ea.png">

<img width="1334" alt="Screen Shot 2021-11-30 at 6 48 09 PM" src="https://user-images.githubusercontent.com/3237612/144146244-a319974e-b6ed-47be-8c34-032a82c60258.png">

## Additional context

Things I've tried:

- Adding the environment variables to the controller directly (gets overwritten).
- Adding the environment variables to the `applicationSet` configuration in the ArgoCD yaml instance (get invalid spec error).

I've looked [here at the ArgoCD Spec](https://argocd-operator.readthedocs.io/en/latest/reference/api.html/) but there's no definition for the `applicationSet` field in the ArgoCD resource, unlike grafana, prometheus, etc.

If there's a workaround for this issue please let me know as we're currently stuck using the "app of apps" pattern which has some downsides for us.

**Comment by caseyscarborough** [comment]
As info - I was able to workaround this issue by manually adding the proxy to the repository inside of ArgoCD. This isn't required when using `Application` manifests instead of using `ApplicationSet` manifests.

![image](https://user-images.githubusercontent.com/3237612/144147967-77b6b50d-6d2f-466e-a090-afbf6d19d89f.png)

**Comment by iam-veeramalla** [comment]
Thanks @caseyscarborough for a very detailed explanation. This is a bug and should be fixed.

**Comment by iam-veeramalla** [comment]
fixed in v1.5.0
